### DISCUSSION TITLE
Run WinDynamicDesktop before login (or even before login screen)

### DISCUSSION THREAD

**Discussion by s31z** [t1m0thyj/WinDynamicDesktop#546] [Ideas]
Hello people,

first I have to say I really love the WinDynamicDesktop app, it is one of my favorite extensions for Windows and I also think it should be included in the OS itself (as on MacOS). Thank you @t1m0thyj for the effort you are putting into this!

One thing that is bugging me (and maybe I am just to much of a perfectionist, but anyway...) is that on startup, the wallpaper of the time of the last shutdown is shown and only updated a few seconds after logging in.
This is completely understandable, as the app is closed on shutdown and restarted only after login is completed.

Still it is bugging me and I would love to already see the correct wallpaper on the login screen 😄

My first idea was to just create a Windows service which calls the WinDynamicDesktop.exe at startup, but after thinking about it, this probably won't work as the app is run by a system user and not the necessary user account then.
Unfortunately, I do not have a lot of experience in writing Windows / .NET applications and therefore cannot really estimate how hard it would be to implement this. 

Maybe some more experienced people can share their ideas here, I would highly appreciate it 🙇‍♂️

Thank you!
Seb

P.S.: When trying to create a Windows service from the app that runs on startup, I get the following error:
```
Application: WinDynamicDesktop.exe
CoreCLR Version: 6.0.1623.17311
.NET Version: 6.0.16
Description: The process was terminated due to an unhandled exception.
Exception Info: System.InvalidOperationException: Showing a modal dialog box or form when the application is not running in UserInteractive mode is not a valid operation. Specify the ServiceNotification or DefaultDesktopOnly style to display a notification from a service application.
   at System.Windows.Forms.Form.ShowDialog(IWin32Window owner)
   at System.Windows.Forms.Form.ShowDialog()
   at WinDynamicDesktop.Localization.SelectLanguage(Boolean exitOnCancel)
   at WinDynamicDesktop.Localization.Initialize()
   at WinDynamicDesktop.AppContext..ctor(String[] args)
   at WinDynamicDesktop.Program.Main(String[] args)
```

**Comment by t1m0thyj** [comment]
Hello, the "Install/Uninstall Startup Task" available in the WDD-scripts repo may do what you want: https://github.com/t1m0thyj/WDD-scripts/tree/master/other#readme

Installing this script registers a task in Task Scheduler that runs at logon, before your PC is unlocked and Windows runs startup programs. I've been hesitant to make this the default behavior in WinDD since it requires admin rights and may not work well for everyone. But in your case it may work better than creating a service.
